### DISCUSSION TITLE
Is it possible to calculate absolute binding free energy with PyAutoFEP?

### DISCUSSION THREAD

**Discussion by Fairly** [luancarvalhomartins/PyAutoFEP#44] [Q&A]
Hi there, thank you for maintaining this awesome repo! I'm wondering whether it is possible to calculate absolute binding free energy with PyAutoFEP, either with the current code base or some modifications on it.

**Comment by luancarvalhomartins** [comment]
(I moved the issue to a discussion, as I believe it is more suitable here. Feel free to open a new one if you think otherwise.)

In principle, using the regular code, no. Because of the way dual-topologies are generated, a molecule must be perturbed into another, so the free energy will be relative.

However, I believe `prepare_dual_topology.py` could be used to perturb a molecule in a version of itself with all non-bonded interactions turned off. Therefore, it might be possible to use it to prepare the systems for running the legs named _ΔG prot elec+vdw_ and _ΔG solv elec+vdw_ in [Aldeghi 2016](https://pubs.rsc.org/en/content/articlelanding/2016/sc/c5sc02678d) (Fig 2 below). But here comes the catch: the current code won't allow to perturb both Coulomb and VdW parameters of atoms at once. I am currently working on a updated version that - for other reasons - will allow such.

![image](https://user-images.githubusercontent.com/29309798/165308209-07654d8a-ba4b-4e86-be2d-5ecd5e5295ae.png)
(Fig 2 from [Chem. Sci., 2016,7, 207-218](https://pubs.rsc.org/en/content/articlelanding/2016/sc/c5sc02678d))

So, the complete answer is: not currently, it will likely be possible in the near future with some tweaking and some (minor?) coding. Furthermore, I think that this would be a quite useful addition to PyAutoFEP and a somewhat low hanging fruit - after the new code allowing VdW+Coulomb perturbation of constant region is deployed. I would be interested in adding it, specially in case there are people interesting in helping code and test.

**Reply by Fairly** [reply]
Thank you for your reply. I will be glad to help coding whenever possible. Only one question, as you mentioned:
"But here comes the catch: the current code won't allow to perturb both Coulomb and VdW parameters of atoms at once."

Is it reasonable to control the Coulomb and VdW terms just with the lambdas.ini configuration?

**Reply by luancarvalhomartins** [reply]
Sorry, I should have unpacked this point. Currently, the terms in the `lambda.ini` are only applyed to the A and B regions of the dual-topology. In case one is perturbing a molecule in a version of itself, all atoms will match so that the entire molecule would be the "constant" region (grey atoms in the fig below). For this atoms, the current version only allows perturbing the Coulomb terms. That's because PyAutoFEP expect that the atomtypes are the same. However, for charged perturbations, pertubring both VdW and Coul terms at once is required to morph an water into an ion. So, when the charged perturbations code makes it in to the tree, allowing `prepare_dual_topology.py` to perturb a molecule should be simple.

![image](https://user-images.githubusercontent.com/29309798/165503864-38adecdd-3603-4490-8df9-7fa3387b6b7a.png)

PS: in case you are asking whether there should be another way to passing a lambda scheme to `prepare_dual_topology.py`, there is. See documentation for the `lambda_input` option.

**Reply by Fairly** [reply]
Oh, got it. Thank you so much for making things clear. Things seem complicated, but please let me know if any thing I could help you with.

**Reply by luancarvalhomartins** [reply]
I would love your (and others) help in implementing and/or testing this. I'm opening a discussion about absolute FE implementation after the code for charged perturbations makes it into the master branch.

**Reply by spadavec** [reply]
hello @luancarvalhomartins -- just curious, is this repo now in support of ABFE calculations?

**Reply by luancarvalhomartins** [reply]
Hi, @spadavec. Thanks for asking.

No, unfortunately, not yet. The code to allow perturbing VdW and lambda that I mentioned above is ready but not yet committed. I plan to do that together with the charged-perturbation code, which I wrote but was not able to test extensively yet.

After that, it could be possible to implement ABFE as in [Chem. Sci., 2016,7, 207-218](https://pubs.rsc.org/en/content/articlelanding/2016/sc/c5sc02678d) with a moderate amount of coding.

As I mentioned above, it would be awesome if other were willing to help in coding and/or testing it :)
